src/constraints.jl

```diff
@@ -0,0 +1,533 @@
+    structure,
+    name::String,
+    extra...
+)
+    for arg in extra
+        _error("Unrecognized argument `$arg`.")
+    end
+end
+
+"""
+    disjunction(
+        model::JuMP.Model, 
+        structure, 
+        name::String = ""; 
+        nested::Bool = false,
+        indicator::LogicalVariableRef
+    )
+
+Function to add a [`Disjunction`](@ref) to a [`GDPModel`](@ref).
+"""
+function disjunction(
+    model::JuMP.Model, 
+    structure, 
+    name::String = ""; 
+    nested::Bool = false,
+    indicator::LogicalVariableRef
+) # TODO add kw argument to build exactly 1 constraint
+    if nested
+        return _disjunction(error, model, structure, name, DisjunctConstraint)
+    else
+        return _disjunction(error, model, structure, name, DisjunctConstraint(indicator))
+    end
+end
+
+################################################################################
+#                              LOGICAL CONSTRAINTS
+################################################################################
+JuMP.operator_to_set(_error::Function, ::Val{:⟹}) = _error(
+    "Cannot use ⟹ in a MOI set (invalid right-hand side). If you are seeing this error, " *
+    "you likely added a logical constraint of the form A ⟹ B == true. " *
+    "Instead, you should enclose the constraint function in parenthesis: " *
+    "(A ⟹ B) == true."
+)
+JuMP.operator_to_set(_error::Function, ::Val{:⇔}) = _error(
+    "Cannot use ⇔ in a MOI set (invalid right-hand side). If you are seeing this error, " *
+    "you likely added a logical constraint of the form A ⇔ B == true. " *
+    "Instead, you should enclose the constraint function in parenthesis: " *
+    "(A ⇔ B) == true."
+)
+
+"""
+    function JuMP.build_constraint(
+        _error::Function, 
+        func::AbstractVector{T},
+        set::MOISelector
+    ) where {T <: Union{LogicalVariableRef, _LogicalExpr}}
+
+Extend `JuMP.build_constraint` to add logical cardinality constraints to a [`GDPModel`](@ref). 
+This in combination with `JuMP.add_constraint` enables the use of 
+`@constraint(model, [name], logical_expr in set)`, where set can be either of the following
+cardinality sets: `AtLeast(n)`, `AtMost(n)`, or `Exactly(n)`.
+
+## Example
+
+To select exactly 1 logical variable `Y` to be `true`, do 
+(the same can be done with `AtLeast(n)` and `AtMost(n)`):
+
+```jldoctest
+julia> model = GDPModel();
+julia> @variable(model, Y[i = 1:2], LogicalVariable);
+julia> @constraint(model, [Y[1], Y[2]] in Exactly(1));
+```
+
+    JuMP.build_constraint(
+        _error::Function, 
+        func::_LogicalExpr,
+        set::_MOI.EqualTo{Bool}
+    )
+
+Extend `JuMP.build_constraint` to add logical propositional constraints to a [`GDPModel`](@ref). 
+This in combination with `JuMP.add_constraint` enables the use of 
+`@constraint(model, [name], logical_expr == true/false)` to define a Boolean expression that must
+either be true or false.
+"""
+# Cardinality logical constraint
```

Comment:
This comment needs to be removed in order for the docstring to register with the function